Memory_Management/Paging.md

```diff
@@ -13,13 +74,91 @@ To implement paging, is strongly reccomended to have implemented exceptions too.
+|-------|-------|------------|-----------------------|-----------|
+|**EXB**| Avail | _Reserved must be 0_ | **Page Base Address** | Available |
+
+
+| 8     | 7       | 6      | 5     |  4      |  3      |  2      |  1      | 0     |
+|-------|---------|--------|-------|---------|---------|---------|---------|-------|
+| **G** | **PAT** | **D**  | **A** | **PCD** | **PWT** | **U/S** | **R/W** | **P** |
+
+
+In this table there are 3 new bits (D, PAT, G) and the Page Base Address as already explained is not pointing to a table but to a memory area. 
+
+In the next section we will se all the fields of an entry.
+
+### Page Table entries fields
+
+Below is a list of all the fields present in the table entries, with an explanation of the most commonly used.
+
+* **P** (Present): If set this tells the CPU that the current page or page table pointed by this entry is currently loaded in physical memory, so when accessed the address translation can be carried out. If is 0 this means that the page is not loaded into memory so a virtual address that contains this entry will cause a Page Fault.
+* **R/W** (Read/Write): If set the page can be both read and written, if clear is in read only mode. When this bit is set in a page Directory it tells the cpu that all its entries will share that setting
+* **User/Supervisor** It describe the privilege level, if clear the page has the Supervisor level, while if it is set the level is Supervisor.
+* **PWT** (Page Level Write Through): controls the caching policy (writhe through or write back), i usually leave it to 0, for more information refer to the Intel Developer Manuals
+* **PCD** (Page Level Cache Disable): controls the caching of individual pages or tables, i usually leave it to 0, for more information refer to the Intel Developer Manuals
+* **A** (Accessed): This value is set by the CPU, if is 0 it means the page hasn't  been accessed yet. Is set when the page (or page teable) have been accessed at least once
+* **D** (Dirty): Indicates if a page has been writtent to when set. This flag applies only to Page Tables. This flag and the accessed flag are provided for being use by the memory management software, the CPU only set it when it's value is 0. Otherwise is up to the Memory Maanger to decide if it has to be cleared or not.
+* **PS** (Page Size): Used only on Page Directory Level, if set it indicates that the current entry point to a 2MB Page, if it is clear it means that the current entry is a 4kb page
+* **PAT** (Page Attribute Table Index): It selects the PAT entry, refer to the Intel Manual for a more detailed explanation
+* **G** (Global) it requires the PGE bit set in in CR4, if set it indicates that when CR3 is loaded or a task switch occurs that page-table or page directory is not invalidated.
+* **EXB** is the Execute Disable bit, available only if supported by the CPU, otherwise is reserved. Refer to the intel manual for this bit.
```

Comment:
usually called XD or NX. Support must be checked for with cpuid, and then it must be enabled in EFER before use.I like to call it XD because it looks like a happy face haha xD